packages/mobile/src/store/wallet-connect/sagas/connectNewWalletSaga.ts

```diff
@@ -47,6 +49,14 @@ function* connectNewWalletAsync(action: ConnectNewWalletAction) {
       const connectData = decryptPayload(data, nonce, sharedSecretDapp)
       const { session, public_key } = connectData
 
+      const isNewWallet = yield* checkIsNewWallet(public_key, Chain.Sol)
+      if (!isNewWallet) return
+
+      const { balance, collectibleCount } = yield* getWalletInfo(
+        public_key,
+        Chain.Sol
```

Comment:
This will need to be cased right?

Reply:
would it for phantom?

Reply:
nope sorry, i didn't expand above lines. this is in the switch